hw09

Jasurr opened the pull request:
feat: implemented classic Spring MVC web application with Thymeleaf and full CRUD for books

- Added entity persistence using database with existing DAO/repository structure
- Implemented UI using Thymeleaf templates and classic Spring Controllers
- Enabled full CRUD operations for the Book entity (view, add, update, delete)
- Added @WebMvcTest-based unit tests for book controllers using mocked services
- Implemented REST API for managing comments (get, add, update, delete)
- Refactored and structured the project; all controllers are covered by tests

Jasurr committed:
feat: implemented classic Spring MVC web application with Thymeleaf and full CRUD for books

- Added entity persistence using database with existing DAO/repository structure
- Implemented UI using Thymeleaf templates and classic Spring Controllers
- Enabled full CRUD operations for the Book entity (view, add, update, delete)
- Added @WebMvcTest-based unit tests for book controllers using mocked services
- Implemented REST API for managing comments (get, add, update, delete)
- Refactored and structured the project; all controllers are covered by tests

Jasurr committed:
feat: implemented classic Spring MVC web application with Thymeleaf and full CRUD for books

- Added entity persistence using database with existing DAO/repository structure
- Implemented UI using Thymeleaf templates and classic Spring Controllers
- Enabled full CRUD operations for the Book entity (view, add, update, delete)
- Added @WebMvcTest-based unit tests for book controllers using mocked services
- Implemented REST API for managing comments (get, add, update, delete)
- Refactored and structured the project; all controllers are covered by tests

stvort commented on hw09/src/main/java/ru/otus/hw/controllers/BookController.java:
```diff
@@ -0,0 +1,76 @@
+        return "redirect:/book";
+    }
+
+    @GetMapping("/delete/{id}")
```
Мы договаривались не делать удаление через GET, он для получения данных

stvort commented on hw09/src/main/java/ru/otus/hw/controllers/CommentController.java:
```diff
@@ -0,0 +1,49 @@
+    private final CommentService commentService;
+
+    @GetMapping("/{id}")
+    @ResponseBody
```
Текущее задание не предусматривает работу c REST, только классический подход. REST будет в следующей работе и там для того, чтобы вернуть сущность в теле ответа нужно будет использовать `@RestController`, у которого `@ResponseBody` внутри. Переделывать методы классического контроллера в методы REST-контроллера, не нужно

Jasurr committed:
Add lesson objectives for MVC and Spring MVC:
- Understand and differentiate MVC and Spring MVC architectures
- Create simple classic web applications using Spring MVC and Spring Boot
- Develop REST services using Spring MVC and Spring Boot

Jasurr committed:
Add lesson objectives for MVC and Spring MVC:
- Understand and differentiate MVC and Spring MVC architectures
- Create simple classic web applications using Spring MVC and Spring Boot
- Develop REST services using Spring MVC and Spring Boot

Commit:
- Understand and differentiate MVC and Spring MVC architectures
- Create simple classic web applications using Spring MVC and Spring Boot
- Develop REST services using Spring MVC and Spring Boot

Commit:
Edited book validation and fixed @WebMvcTest issues

Commit:
- Understand and differentiate MVC and Spring MVC architectures
- Create simple classic web applications using Spring MVC and Spring Boot
- Develop REST services using Spring MVC and Spring Boot

Commit:
Merge branch 'hw09' of https://github.com/Jasurr/2025-01-otus-spring-nishonov into hw09

Commit:
Add transaction rollback for testing Author and Genre services, update Book model

stvort commented on hw09/src/main/java/ru/otus/hw/controllers/BookController.java:
```diff
@@ -0,0 +1,80 @@
+import java.util.stream.Collectors;
+
+@Controller
+@RequestMapping("/book")
```
Мы обсуждали, что лучше писать весь маппинг над методами. Так не приходится прыгать между классом и методом + можно найти эндпойнт поиском
